### PR TITLE
Fix Geometry UOs and Boundary Marker

### LINKED ISSUE #24645: Enable Geometry User Objects for blocks
## Reason

When refining lower dimensional subdomains, such as secondary surface meshes for mortar contact, it would be convenient to apply a Geometry User Object to the lower D _block_ rather than a sideset.

Geometry User Objects will "snap" a selected set of nodes to a user defined geometry. For example `GeometrySphere` can be used to place new nodes added by mesh refinement to the surface of a perfect sphere/circle.

## Design
In `GeometryBase::meshChanged()` we need to iterate over selected subdomain nodes in addition to boundary nodes.

## Impact
Added capability

### PR DESCRIPTION
**Pull request by dschwen**
Closes #24645

### CODE REVIEW

File: framework/src/markers/BoundaryMarker.C [hunk 1]
```diff
@@ -43,23 +44,26 @@ BoundaryMarker::computeElementMarker()
...
-
+    for (const auto boundary : _boundary_ids)
+    {
+      const auto it = _bnd_elem_ids.find(boundary);
+      if (it != _bnd_elem_ids.end())
+        for (const auto id : it->second)
+        {
+          const auto elem = _mesh.elemPtr(id);
+          const auto r = _current_elem->vertex_average() - elem->vertex_average();
+          if (r.norm() < _distance)
```
Comment (lindsayad) [inline framework/src/markers/BoundaryMarker.C:63]:
```suggestion
          if ((elem->id() == _current_elem->id()) || (r.norm() < _distance))
```

File: framework/src/userobjects/GeometryBase.C [hunk 2]
```diff
@@ -58,4 +63,19 @@ GeometryBase::meshChanged()
...
+  {
+    // check if node is part of any of the selected blocks
+    const auto & node_blocks = _mesh.getNodeBlockIds(**node);
+    for (const auto subdomain_id : _subdomain_ids)
+      if (node_blocks.count(subdomain_id))
+      {
+        snapNode(**node);
+        break;
+      }
+  }
```
Comment (lindsayad) [inline framework/src/userobjects/GeometryBase.C:80]: Should we be checking whether blocks are lower dimensional? Maybe I just need to know what we're snapping to. I guess in my brain I'm visualizing snapping all the nodes of a QUAD or TRI onto a line which doesn't sound good

File: test/tests/markers/boundary_marker/multiple.i [hunk 3]
```diff
@@ -0,0 +1,44 @@
+###########################################################
+# This is a test of the Mesh Marker System. It marks
+# elements with flags indicating whether they should be
+# refined, coarsened, or left alone. This system
+# has the ability to use the Mesh Indicator System.
+#
+# @Requirement F2.50
+###########################################################
```
Comment (lindsayad) [inline test/tests/markers/boundary_marker/multiple.i:8]: I don't think we need this?

**Review by lindsayad — CHANGES_REQUESTED** [review]

File: framework/src/userobjects/GeometryBase.C (hunk 2 above)
Comment (dschwen) [reply framework/src/userobjects/GeometryBase.C:80]: Yeah, I was thinking about that, too, but the API does permit snapping to a volume, not just a surface. This you enforce  proper mesh biasing upon refinement for example.

File: framework/src/markers/BoundaryMarker.C (hunk 1 above)
Comment (dschwen) [reply framework/src/markers/BoundaryMarker.C:69]: I did something along those lines

File: test/tests/markers/boundary_marker/multiple.i (hunk 3 above)
Comment (dschwen) [reply test/tests/markers/boundary_marker/multiple.i:8]: Forgot to remove it. Is that our old "SQA system"?

**Review by lindsayad — APPROVED** [review]

**Pull request merged**